src/pymovements/dataset/dataset_files.py

```diff
@@ -318,16 +272,75 @@ def load_gaze_file(
+            gaze_df = from_csv(
+                filepath,
+                trial_columns=definition.trial_columns,
+                add_columns=add_columns,
+                column_dtypes=definition.filename_format_dtypes,
+            )
+
+            pixel_columns: list[str] = list(
+                set(GazeDataFrame.valid_pixel_position_columns) & set(gaze_df.frame.columns),
+            )
+            position_columns: list[str] = list(
+                set(GazeDataFrame.valid_position_columns) & set(gaze_df.frame.columns),
+            )
+            velocity_columns: list[str] = list(
+                set(GazeDataFrame.valid_velocity_columns) & set(gaze_df.frame.columns),
+            )
+            acceleration_columns: list[str] = list(
+                set(GazeDataFrame.valid_acceleration_columns) & set(gaze_df.frame.columns),
+            )
+
+            column_specifiers: list[list[str]] = []
+            if len(pixel_columns) > 0:
+                gaze_df.nest(pixel_columns, output_column='pixel')
+                column_specifiers.append(pixel_columns)
+
+            if len(position_columns) > 0:
+                gaze_df.nest(position_columns, output_column='position')
+                column_specifiers.append(position_columns)
+
+            if len(velocity_columns) > 0:
+                gaze_df.nest(velocity_columns, output_column='velocity')
+                column_specifiers.append(velocity_columns)
+
+            if len(acceleration_columns) > 0:
+                gaze_df.nest(acceleration_columns, output_column='acceleration')
+                column_specifiers.append(acceleration_columns)
         else:
-            gaze_df = pl.read_csv(filepath, **custom_read_kwargs)
+            gaze_df = from_csv(
+                filepath,
+                experiment=definition.experiment,
+                time_column=definition.time_column,
+                distance_column=definition.distance_column,
+                pixel_columns=definition.pixel_columns,
+                position_columns=definition.position_columns,
+                velocity_columns=definition.velocity_columns,
+                acceleration_columns=definition.acceleration_columns,
+                trial_columns=definition.trial_columns,
+                column_map=definition.column_map,
+                add_columns=add_columns,
+                column_dtypes=definition.filename_format_dtypes,
+                **custom_read_kwargs,
+            )
     elif filepath.suffix == '.feather':
-        gaze_df = pl.read_ipc(filepath)
+        gaze_df = from_ipc(
+            filepath,
+            experiment=definition.experiment,
+            column_map=definition.column_map,
+            add_columns=add_columns,
+            column_dtypes=definition.filename_format_dtypes,
+        )
     elif filepath.suffix == '.asc':
-        gaze_df, _ = parse_eyelink(filepath, **custom_read_kwargs)
+        gaze_df = from_asc(filepath, **custom_read_kwargs)
```

Comment:
`add_columns` and `column_dtypes` should be probably passed to `from_asc` too

Reply:
added the parameter in gaze.io and passed the params in dataset_files.py